tests/unit/containers/tutor-home-page/subjects-step/SubjectsStep.spec.jsx

```diff
@@ -0,0 +1,155 @@
+import { screen, fireEvent, waitFor } from '@testing-library/react'
+import { renderWithProviders } from '~tests/test-utils'
+import SubjectsStep from '~/containers/tutor-home-page/subjects-step/SubjectsStep'
+import { StepProvider } from '~/context/step-context'
+
+import { categoryService } from '~/services/category-service'
+import { subjectService } from '~/services/subject-service'
+
+vi.mock('~/services/category-service')
+vi.mock('~/services/subject-service')
+
+import {
+  initialValues,
+  tutorStepLabels
+} from '~/components/user-steps-wrapper/constants'
+
+const mockCategoriesNames = [
+  { _id: '1', name: 'Category 1' },
+  { _id: '2', name: 'Category 2' },
+  { _id: '3', name: 'Category 3' },
+  { _id: '4', name: 'Category 4' }
+]
+
+const mockSubjectsNames = [
+  { _id: '5', name: 'Subject 1' },
+  { _id: '6', name: 'Subject 2' }
+]
+
+categoryService.getCategoriesNames.mockResolvedValue({
+  data: mockCategoriesNames
+})
+
+subjectService.getSubjectsNames.mockResolvedValue({
+  data: mockSubjectsNames
+})
+
+const btnsBox = (
+  <div>
+    <button>back</button>
+    <button>next</button>
+  </div>
+)
+
+const chooseOption = (name) => {
+  const firstOption = screen.getByRole('option', {
+    name
+  })
+  fireEvent.click(firstOption)
+}
+
+const chooseCategoryOrSubject = async (inputField, value) => {
+  await waitFor(() => {
+    fireEvent.click(inputField)
+    fireEvent.change(inputField, {
+      target: { value }
+    })
+  })
+
+  chooseOption(value)
+}
+
+describe('SubjectsStep test with some data', () => {
+  beforeEach(async () => {
+    await waitFor(() => {
+      renderWithProviders(
+        <StepProvider
+          initialValues={initialValues}
+          stepLabels={tutorStepLabels}
+        >
+          <SubjectsStep btnsBox={btnsBox} stepLabel={'subjects'} />
+        </StepProvider>
+      )
+    })
+  })
+
+  afterEach(() => {
+    vi.clearAllMocks()
+  })
+
+  it('should render SubjectsStep title', () => {
+    const title = screen.getByText('becomeTutor.categories.title')
+    expect(title).toBeInTheDocument()
+  })
+
+  it('should add a new subject and delete a subject', async () => {
+    const addSubject = screen.getByTestId('add-subject')
+    expect(addSubject).toBeInTheDocument()
+
+    const firstField = screen.getByLabelText(
```

Comment:
Variable names might be something like `categoryField' instead of `firstField' to make the code a bit easier to read. The same thing works for the subject field.

Reply:
I agree, the variables' names were changed to `categoryField` and `subjectField`.